server/src/main/java/org/opensearch/search/sandboxing/cancellation/MostResourceIntensiveTaskFirst.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.sandboxing.cancellation;
+
+import org.opensearch.search.sandboxing.SandboxResourceType;
+import org.opensearch.tasks.Task;
+
+import java.util.Comparator;
+
+public class MostResourceIntensiveTaskFirst extends AbstractTaskCancellationStrategy {
+  private final SandboxResourceType resourceType;
```

Comment:
We might have multiple resources in the future so we should keep this flexible and open to add more resource types.